### PR TITLE
Queues: convert to Python library

### LINKED ISSUE #2186: Queues: convert to Python library
Our queue frontend is getting larger and larger, and is totally polluting the `calyx-py` directory. Let us move it all into its own directory, and in doing so let us convert it into a Python library. Doing so will also make imports easier. In doing all this, let us make sure that the queues still get tested via `runt`. At present they get tested somewhat automatically because they live under `calyx-py/test/correctness/...`.

### PR DESCRIPTION
**Pull request by polybeandip**
Closes #2186 and the test harness part of #2191.

More specifically, this PR does the following:
- set up separate directories for our package, data file generation, and tests
- move each queue's python generator and associated main component to different files: main components are in `test/`
- remove PCQs from `data_gen/gen_queue_data_expect.sh` (for now)
- tweak runt stanzas to work with new file structure
- tweak github actions to install `queues` for CI and run `gen_queue_data_expect.sh`
- gitignore large `.data` and `.expect` files
- fill in README for our library
- update links in the docs to our new file locations and add install instructions

### CODE REVIEW

**Comment by polybeandip** [comment]
@anshumanmohan is it okay for me to move `sdn.py` (and it's associated data and expect file)?

I think I remember you saying these files were artifacts for a paper. For this reason, I wanted to check if it's okay for me to move them around and tweak their associated `runt` stanza (not the `cmd` part, only `path`).

**Comment by anshumanmohan** [comment]
Yay, thank you! Been looking forward to these changes haha. I will ask @csziklai and @KabirSamsi to glance over this PR and offer their own reviews if they’d like. It’s a great way to stay in the loop. Regardless, I’ll go through and review it tomorrow EoD

**Comment by anshumanmohan** [comment]
Yes, that’s okay re: SDN! I’d like to resuscitate it and have it hang around _somewhere_, so just let me know where it ends up

**Comment by polybeandip** [comment]
You can find it at [`frontends/queues/tests/sdn_test.py`](https://github.com/calyxir/calyx/blob/48cd327d949cc06f976b7cea5254df83c9824d8a/frontends/queues/tests/sdn_test.py).

I've made sure to alter the SDN runt stanzas so they point to this new location now. Those tests still pass just fine!

**Review by csziklai — APPROVED** [review]

**Comment by EclecticGriffin** [comment]
wowowowow, this might be a record for deleted lines!

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -161,14 +161,21 @@ jobs:
...
         cd -
         cd frontends/mrxl
         FLIT_ROOT_INSTALL=1 flit install --symlink
+        cd -
+        cd frontends/queues
+        FLIT_ROOT_INSTALL=1 flit install --symlink
+
+    - name: Generate queue .data and .expect files
+      working-directory: /home/calyx
+      run: ./frontends/queues/test_data_gen/gen_test_data.sh
```
Comment (anshumanmohan) [inline .github/workflows/rust.yml:178]: 🎉

File: docs/frontends/queues.md [hunk 2]
```diff
@@ -189,21 +195,20 @@ It uses a counter `i` and instantiates, in turn, a binary heap that accepts 64-b
 The source code is available in [`stable_binheap.py`][stable_binheap.py].
 
 [builder]: ../builder/calyx-py.md
```
Comment (anshumanmohan) [inline docs/frontends/queues.md:197]: Thanks a lot for doing a careful job with the docs! Historically we have a bad habit of changing the file structure but ignoring the docs, leading to a buncha broken links!

File: frontends/queues/queues/strict_or_rr.py [hunk 3]
```diff
@@ -1,16 +1,7 @@
 # pylint: disable=import-error
-import os
-import sys
-import inspect
-
-currentdir = os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
-parentdir = os.path.dirname(currentdir)
-sys.path.insert(0, parentdir)
```
Comment (anshumanmohan) [inline frontends/queues/queues/strict_or_rr.py:8]: Gorgeous

**Review by anshumanmohan — APPROVED** [review]
Wowowow thanks for this massive massive cleanup! 👏 It's really looking a lot more organized and maintainable. I have reviewed this and learned a bunch from the process. Merge at will!

**Pull request merged**